files/src/main/java/com/fincity/saas/files/dao/FilesSecuredAccessKeyDao.java

```diff
@@ -0,0 +1,41 @@
+package com.fincity.saas.files.dao;
+
+import static com.fincity.saas.files.jooq.tables.FilesSecuredAccessKey.FILES_SECURED_ACCESS_KEY;
+
+import org.jooq.types.ULong;
+import org.springframework.stereotype.Service;
+
+import com.fincity.saas.commons.jooq.dao.AbstractDAO;
+import com.fincity.saas.files.dto.FilesSecuredAccessKey;
+import com.fincity.saas.files.jooq.tables.records.FilesSecuredAccessKeyRecord;
+
+import reactor.core.publisher.Mono;
+
+@Service
+public class FilesSecuredAccessKeyDao extends AbstractDAO<FilesSecuredAccessKeyRecord, ULong, FilesSecuredAccessKey> {
+
+	protected FilesSecuredAccessKeyDao() {
+		super(FilesSecuredAccessKey.class, FILES_SECURED_ACCESS_KEY, FILES_SECURED_ACCESS_KEY.ID);
+	}
+
+	public Mono<FilesSecuredAccessKey> getAccessByKey(String key) {
+
+		return Mono.from(
+
+		        this.dslContext.selectFrom(FILES_SECURED_ACCESS_KEY)
+		                .where(FILES_SECURED_ACCESS_KEY.ACCESS_KEY.eq(key))
+		                .limit(1))
+		        .map(e -> e.into(FilesSecuredAccessKey.class));
+
+	}
+
+	public Mono<Boolean> setAccessCount(ULong id, ULong count) {
```

Comment:
It should be an increasing count, and the query has to be counted + 1 in the update. and the condition has to be where id = given id and accessed_count < limit. That way you can guarantee the concurrency.

Reply:
Updated the query with mentioned comment.